lib/solid/tag/render.ex

```diff
@@ -52,7 +53,12 @@ defmodule Solid.Tag.Render do
     {file_system, instance} = options[:file_system] || {Solid.BlankFileSystem, nil}
 
     template_str = file_system.read_template_file(template, instance)
-    template = Solid.parse!(template_str, options)
+
+    cache_key = :md5 |> :crypto.hash(template_str) |> Base.encode16(case: :lower)
+
+    template =
+      apply(cache_module, :fetch, [cache_key, fn -> Solid.parse!(template_str, options) end])
```

Comment:
Hmm I think we want to fix this to not use `Solid.parse!` but `Solid.parse` and call `Solid.Context.put_errors` instead of raising like it does today.

And ensure that the result is not cached if it failed to parse.

We could change the behaviour to have two callbacks

```elixir
@callback get(key :: term) :: {:ok, Template.t()} | {:error, :not_found}

@callback put(key :: term, Template.t()) :: :ok | {:error, term}
```

Then we would only `put` if the template succeeded to parse.

WDYT?

Reply:
sounds good :) I will change that